src/bun.js/bindings/highway_strings.cpp

```diff
@@ -0,0 +1,790 @@
+        const auto mask_gt_max = hn::Gt(text_vec, vec_max_ascii);
+
+        const auto found_mask = hn::Or(
+            hn::Or(mask_quote, mask_backslash),
+            hn::Or(mask_lt_min, mask_gt_max));
+
+        const intptr_t pos = hn::FindFirstTrue(d, found_mask);
+        if (pos >= 0) {
+            return i + pos;
+        }
+    }
+
+    for (; i < text_len; ++i) {
+        const uint8_t c = text[i];
+        if (c == quote || c == '\\' || (c < 0x20 || c > 0x7E)) {
+            return i;
+        }
+    }
+
+    return text_len;
+}
+
+size_t IndexOfNewlineOrNonASCIIOrHashOrAtImpl(const uint8_t* HWY_RESTRICT start_ptr, size_t search_len)
+{
+    ASSERT(search_len > 0);
+
+    D8 d;
+    const size_t N = hn::Lanes(d);
+
+    const auto vec_hash = hn::Set(d, '#');
+    const auto vec_at = hn::Set(d, '@');
+    const auto vec_min_ascii = hn::Set(d, uint8_t { 0x20 });
+    const auto vec_max_ascii = hn::Set(d, uint8_t { 0x7E });
+
+    size_t i = 0;
+    const size_t simd_text_len = search_len - (search_len % N);
+    for (; i < simd_text_len; i += N) {
+        const auto vec = hn::LoadU(d, start_ptr + i);
+
+        const auto mask_hash = hn::Eq(vec, vec_hash);
+        const auto mask_at = hn::Eq(vec, vec_at);
+        const auto mask_lt_min = hn::Lt(vec, vec_min_ascii);
+        const auto mask_gt_max = hn::Gt(vec, vec_max_ascii);
+
+        const auto found_mask = hn::Or(hn::Or(mask_hash, mask_at), hn::Or(mask_lt_min, mask_gt_max));
+
+        const intptr_t pos = hn::FindFirstTrue(d, found_mask);
+        if (pos >= 0) {
+            return i + pos;
+        }
+    }
+
+    for (; i < search_len; ++i) {
+        const uint8_t char_ = start_ptr[i];
+        if (char_ == '#' || char_ == '@' || char_ < 0x20 || char_ > 127) {
+            return i;
+        }
+    }
+
+    return search_len;
+}
+
+size_t IndexOfNewlineOrNonASCIIImpl(const uint8_t* HWY_RESTRICT start_ptr, size_t search_len)
+{
+    ASSERT(search_len > 0);
+
+    D8 d;
+    const size_t N = hn::Lanes(d);
+
+    // SIMD constants
+    const auto vec_max_ascii = hn::Set(d, uint8_t { 127 });
+    const auto vec_min_ascii = hn::Set(d, uint8_t { 0x20 });
+
+    // FUTURE TODO: normalize tabs
+    // Some tests involving githubactions depend on tabs not being normalized right now.
+
+    size_t i = 0;
+    const size_t simd_text_len = search_len - (search_len % N);
+    // Process full vectors
+    for (; i < simd_text_len; i += N) {
+        const auto vec = hn::LoadU(d, start_ptr + i);
+        const auto mask_lt_min = hn::Lt(vec, vec_min_ascii);
+        const auto mask_gt_max = hn::Gt(vec, vec_max_ascii);
+
+        const auto found_mask = hn::Or(mask_gt_max, mask_lt_min);
+
+        const intptr_t pos = hn::FindFirstTrue(d, found_mask);
+        if (pos >= 0) {
+            return i + pos;
+        }
+    }
+
+    // Scalar check for the remainder
+    for (; i < search_len; ++i) {
+        const uint8_t char_ = start_ptr[i];
+        if (char_ > 127 || char_ < 0x20) {
```

Comment:
```suggestion
        if (char_ > 127 || char_ == '\n' || char_ == '\r') {
```
originally this check was `if (char_ > 127 || char_ < 0x20 || char_ == '\n' || char_ == '\r') {`. is it correct to accept anything less than a space?

Reply:
< 0x20 covers "\n" and "\r"